ci/format_pre.sh

```diff
@@ -44,6 +44,10 @@ bazel run "${BAZEL_BUILD_OPTIONS[@]}" //configs:example_configs_validation
 CURRENT=flake8
 bazel run "${BAZEL_BUILD_OPTIONS[@]}" //tools/code_format:python_flake8 "$(pwd)"
 
+CURRENT=yamllint
+bazel run "${BAZEL_BUILD_OPTIONS[@]}" //tools/code_format:yaml_lint "$(pwd)"
```

Comment:
I meant to suggest this on the previous PR around whitespace, but before we do more here, can we consolidate as many checks here with https://github.com/envoyproxy/envoy/blob/main/support/hooks/pre-push? Many devs run that pre push script which catches issues before it gets to CI. Can we do YAML linting, glint, etc. on a per-file basis in that script also? I'm not sure we can do all checks, but I think we can do most of them and it saves people a huge amount of time. This is also why it's important to have "fix" targets to also not annoy people and auto fix where possible.

Reply:
yep - we need to use diffs because they are very resource consuming

re autofix - unless we dont use the definition dicts then its not so easy to autofix yaml - when its parsed there is no way to discriminate whether its indented code, or a dictionary - so it would change lots of files, and enforce a style we might not want

yep agreed re hooks, but it needs in that case to work on diffs, which is doable but more complex

ill look now at adding

i also want to flag the large amount of yaml changes, which i think is important

Ive done it mostly by hand, checking etc, so i think its good - but, more eyes, etc..

Reply:
Yeah I will go through all the files later, but let's go ahead and try to make things work on the pre-push hook in the interim.

Reply:
cool - i think it would make sense to do that separately - i think i can do it with git + yamllint - it probs doesnt make sense to add git as a py dependency right now i think - and i guess it can skip yamllint if its not installed

Reply:
there might be an autofix - ill investigate further - https://github.com/adrienverge/yamllint/issues/62#issuecomment-679886759